framework/src/util/Utils.cpp

```diff
@@ -90,8 +110,35 @@ std::string library_suffix()
 
 }
 
+
 namespace cppmicroservices {
+  //-------------------------------------------------------------------
+  // Unicode Utility functions
+  //-------------------------------------------------------------------
+#ifdef US_PLATFORM_WINDOWS
+
+  // function returns empty string if inStr is empty or if the conversion failed
+  std::wstring ToWString(const std::string& inStr)
+  {
+    int wchar_count = MultiByteToWideChar(CP_UTF8, 0, inStr.c_str(), -1, NULL, 0);
+    std::unique_ptr<wchar_t[]> wBuf(new wchar_t[wchar_count+1]);
```

Comment:
Yes, the extra character is to accomodate the case when wchar_count == 0 (when the api call fails). When the API succeeds we set the extra character at the end with '\0' so the string ends with "\0\0". I was trying to keep the code simple (without another if clause or throwing an exception) at the cost of one character per function call. 

I like the vector solution Krish suggested, however, that solution failed in MinGW with SEG_FAULT. If throwing an exception makes sense, the extra character can be removed.